packages/react/src/Input/Input.tsx

```diff
@@ -12,12 +12,19 @@ export interface InputProps extends React.ComponentPropsWithoutRef<'input'> {
 }
 
 export const Input = forwardRef<HTMLInputElement, InputProps>((props, ref) => {
-  const { className, isInvalid, size, prefix, as, type, ...rest } = props;
+  const {
+    className,
+    isInvalid,
+    size,
+    prefix,
+    as,
+    type: typeProp,
+    ...rest
+  } = props;
 
   const Component = as ?? 'input';
 
-  // if type isn't specified, and the component is input, we set it to `text`
-  const inputType = type ?? Component === 'input' ? 'text' : undefined;
```

Comment:
For this to work as intended, parentheses was needed around `Component === 'input' ? 'text' : undefined`. To get rid of that ambiguity, I extracted it out into a function and rewrote it